lib/identity/stateTransitions/assetLock/proof/chain/validateChainAssetLockProofStructureFactory.js

```diff
@@ -0,0 +1,114 @@
+const { Transaction } = require('@dashevo/dashcore-lib');
+const chainAssetLockProofSchema = require('../../../../../../schema/identity/stateTransition/assetLock/proof/chainAssetLockProof.json');
+
+const convertBuffersToArrays = require('../../../../../util/convertBuffersToArrays');
+const InvalidIdentityAssetLockProofHeightError = require('../../../../../errors/InvalidIdentityAssetLockProofHeightError');
+const InvalidIdentityAssetLockProofOutPointError = require('../../../../../errors/InvalidIdentityAssetLockProofOutPointError');
+const InvalidIdentityAssetLockTransactionError = require('../../../../../errors/InvalidIdentityAssetLockTransactionError');
+const IdentityAssetLockTransactionOutputNotFoundError = require('../../../../../errors/IdentityAssetLockTransactionOutputNotFoundError');
+const InvalidIdentityAssetLockTransactionOutputError = require('../../../../../errors/InvalidIdentityAssetLockTransactionOutputError');
+
+/**
+ * @param {JsonSchemaValidator} jsonSchemaValidator
+ * @param {StateRepository} stateRepository
+ * @returns {validateChainAssetLockProofStructure}
+ */
+function validateChainAssetLockProofStructureFactory(
+  jsonSchemaValidator,
+  stateRepository,
+) {
+  /**
+   * @typedef {validateChainAssetLockProofStructure}
+   * @param {RawChainAssetLockProof} rawAssetLockProof
+   */
+  async function validateChainAssetLockProofStructure(
+    rawAssetLockProof,
+  ) {
+    const result = jsonSchemaValidator.validate(
+      chainAssetLockProofSchema,
+      convertBuffersToArrays(rawAssetLockProof),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    const { height, outPoint: outPointBuffer } = rawAssetLockProof;
+
+    if (!await stateRepository.verifyChainLockHeight(height)) {
```

Comment:
You can just use the last header (the function is in the state repo already) to get `coreChainLockedHeight` and compare it with `coreChainLockedHeight` from proofs. If height from header is higher then return with `InvalidIdentityAssetLockProofHeightError`